internal/nmc/helper.go

```diff
@@ -0,0 +1,74 @@
+package nmc
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+)
+
+//go:generate mockgen -source=helper.go -package=nmc -destination=mock_helper.go
+
+type Helper interface {
+	Get(ctx context.Context, name string) (*kmmv1beta1.NodeModulesConfig, error)
+	Create(ctx context.Context, name string) (*kmmv1beta1.NodeModulesConfig, error)
+	GetModuleEntry(nmc *kmmv1beta1.NodeModulesConfig, modNamespace, modName string) *kmmv1beta1.NodeModuleSpec
+	UpsertModuleEntry(nmc *kmmv1beta1.NodeModulesConfig, modNamespace, modName string) *kmmv1beta1.NodeModuleSpec
+}
+
+type helper struct {
+	client client.Client
+}
+
+func NewHelper(client client.Client) Helper {
+	return &helper{
+		client: client,
+	}
+}
+
+func (h *helper) Get(ctx context.Context, name string) (*kmmv1beta1.NodeModulesConfig, error) {
+	nmc := kmmv1beta1.NodeModulesConfig{}
+	err := h.client.Get(ctx, types.NamespacedName{Name: name}, &nmc)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("failed to get NodeModulesConfig %s: %v", name, err)
+	}
+	return &nmc, nil
+}
+
+func (h *helper) Create(ctx context.Context, name string) (*kmmv1beta1.NodeModulesConfig, error) {
+	nmc := kmmv1beta1.NodeModulesConfig{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Spec: kmmv1beta1.NodeModulesConfigSpec{
+			Modules: []kmmv1beta1.NodeModuleSpec{},
+		},
+	}
+	err := h.client.Create(ctx, &nmc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create NodeNodulesState %s: %v", name, err)
+	}
+
+	return h.Get(ctx, name)
```

Comment:
Why do we need `h.Get(ctx, name)`?
```suggestion
	err := h.client.Create(ctx, &nmc)
	if err != nil {
		return nil, fmt.Errorf("failed to create NodeNodulesConfig %s: %v", name, err)
	}
	
	return &nmc, nil
```